WMIAdventure/frontend/src/js/components/battle/organisms/BattleView/BattleView.js

```diff
@@ -62,6 +123,118 @@ class BattleView extends React.Component {
+        });
+    }
+
+    // get proper mini cards to DOM
+    getMiniCards = (enemy) => {
+        return (
+            [...Array(5)].map(
+                (e,i) => {
+                    return (
+                        <MiniCardView key={enemy ? `enemyCard-${i}` : `userCard-${i}`}
+                                      changeCardsOrder={
+                                          enemy ? this.changeCardsEnemyOrder : this.changeCardsUserOrder
+                                      }
+                                      cardIndexInDeck={
+                                          enemy ? this.state.cardsEnemyOrder[i] : this.state.cardsUserOrder[i]
+                                      }
+                                      setTranslateX={
+                                          enemy ? this.state.enemyMiniCardsTranslateX[i]
+                                              : this.state.userMiniCardsTranslateX[i]
+                                      }
+                                      enemy={enemy} user={!enemy}
+                                      cardLevel={this.state.cardLevels[i]}
+                                      animationDuration={`0.${9 - i}`}
+                                      cardImage={this.state.cardIcons[i]}/>
+                    );
+                })
+        );
+    }
+
+    // get proper compact cards to DOM
+    getCompactCards = (enemy) => {
+        return (
+            [...Array(5)].map(
+                (e,i) => {
+                    return (
+                        <CompactCardView key={enemy ? `enemyCompactCard-${i}` : `userCompactCard-${i}`}
+                                         cardIndexInDeck={enemy ? this.state.cardsEnemyOrder[i]
+                                             : this.state.cardsUserOrder[i]}
+                                         cardImage={this.state.cardIcons[i]} cardName={`Karta ${i+1}`}
+                                         setWidth={'124px'} cardLevel={3} setHeight={'200px'}
+                                         setTranslateX={enemy ? this.state.enemyCompactCardTranslateX
+                                             : this.state.userCompactCardTranslateX}
+                                         setTranslateY={enemy ? this.state.enemyCompactCardTranslateY
+                                             : this.state.userCompactCardTranslateY}
+                                         setMargin={enemy ? '0 0 0 10px' : '0 10px 0 0'} />
+                    );
+                })
+        );
+    }
+
+    countTargetEffects(userEffects, userTarget) { // to set correct EffectsIconsContainer gap
+        let effects;
+        if(userEffects) effects = userUsedEffects;
+        else effects = enemyUsedEffects;
+        let enemyCount = 0;
+        let userCount = 0;
+        for(let i=0; i<effects.length; i++) {
+            if(effects[i].target_player === this.state.user) userCount++;
+            else enemyCount++
+        }

```

Comment:
Po co iterować po indexach? 
```
for(const effect of effects)
```
```suggestion
        for (const effect of effects) {
            if (effect.target_player === this.state.user)
                userCount++;
            else
                enemyCount++;
        }
```